src/modules/dual_path.py

```diff
@@ -482,3 +496,141 @@ def evaluate_network(self, set_names, top_down_language_activation=False):
+        weights_identif_role = self.inputs.weights_identif_role['test']
+        event_semantics = self.inputs.event_sem_activations['test']
+        target_lang_act = self.inputs.target_lang_act['test']
+        set_lines = self.inputs.testlines_df
+        num_sentences = self.inputs.num_test
+        directory = self.inputs.directory
+        epoch_range = range(self.starting_epoch, self.epochs)
+        buffer_to_produce = [None] * 2 
+        top_down_language_activation = False
+
+
+        # Only test priming for last epoch
+        epoch = self.epochs
+
+        self.srn.load_weights(set_weights_folder=directory, set_weights_epoch=epoch)
+
+        prime_structure = ''
+        prime_sentence = ''
+        for line in set_lines.itertuples():
+            # Process prime sentence as if training
+            line_idx = line.Index
+            if ((line_idx % 2) == 0):
+                prime_structure = line.sentence_structure
+                prime_sentence = line.target_sentence
+                prime_lang = line.lang
+                self.feed_line(line.target_sentence_idx, target_lang_act[line_idx],
+                               weights_role_concept[line_idx],
+                               weights_concept_role[line_idx],
+                               weights_role_identif[line_idx],
+                               weights_identif_role[line_idx],
+                               event_semantics[line_idx],
+                               backpropagate=True, activate_target_lang=True)
+                
+            # Process target as if testing
+            else:
+                target_pos, target_sentence_idx, target_lang = line.target_pos, line.target_sentence_idx, line.lang
+                alt_sentence_idx, alternative_sentence = line.alt_sentence_idx, line.alt_sentence
+                produced_idx, entropy_idx = self.feed_line(line.target_sentence_idx + buffer_to_produce,
+                                                       target_lang_act[line_idx],
+                                                       weights_role_concept[line_idx],
+                                                       weights_concept_role[line_idx],
+                                                       weights_role_identif[line_idx],
+                                                       weights_identif_role[line_idx],
+                                                       event_semantics[line_idx],
+                                                       activate_target_lang=not top_down_language_activation)
+
+                produced_sentence = self.inputs.sentence_from_indices(produced_idx)
+                produced_pos = self.inputs.sentence_pos(produced_idx)
+                alt_pos = self.inputs.sentence_pos(alt_sentence_idx)
+
+                # We ignore periods. Quite a few sentences without periods are produced that are 
+                # otherwise correct
+                period_idx = self.inputs.df_query_to_idx("pos == '.'")
```

Comment:
Call self.inputs.period_idx instead (the only non-FYI comment). I had also encountered this issue and used to check without the period, but the +2 buffer solved it for me